.github/workflows/update_version_tag.yml

```diff
@@ -0,0 +1,44 @@
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  validate_version:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ inputs.repository }}
+          fetch-depth: 1
+
+      - name: Validate Version
+        shell: ruby {0}
+        run: |
+          version = File.read("VERSION").strip
+          if !version.match?(/\A\d+\z/)
+            puts "VERSION file must contain a single integer"
+            exit 1
+          end
+          puts "VERSION = v#{version}"
+
+  tag_version:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
```

Comment:
We don't need this condition, right?

```suggestion
```

Reply:
Right... I meant to have the validation job run on pull requests as well. Fixed!

Now we do need this condition